cpp/include/cudf/utilities/traits.hpp

```diff
@@ -292,4 +323,45 @@ struct is_simple_impl {
  **/
 constexpr inline bool is_simple(data_type type) { return not is_compound(type); }
 
+/**
+ * @brief Indicates whether `T` is a nesting type.
+ *
+ * "Nesting" types are distinct from compound types in that they
+ * can have an arbitrarily deep list of descendants of the same
+ * type. Strings are not a nested type, but lists are.
```

Comment:
Here you use nested in the same paragraph as nesting.

Reply:
Yes, I wrestled with this as well.   There's two ways to think about it:

- List<int> is _not_ actually nested.  It's just a list of ints.   List<List<int>> _is_ nested because you have a type nested within itself.    In this sense, List is a nest_ing_ type since it can nest.

or

- The opposite.  In other words, because List<> must always contain another type, even if that type is simple, it is by default nested in all cases.  

This is probably just semantic hair splitting.  I'll change to "nested" everywhere :)

Reply:
But lists are always in columns. So you have a column of lists of ints, not just a list of ints. They are called nested types because they utilise column children and offsets.

Reply:
Strings column is a nested type too.